pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/AvoidCallingFinalize.xml

```diff
@@ -88,6 +88,48 @@ public class InputFinalize {
     {
         super.finalize();
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>overloaded finalize is ok</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public void foo() {
+        finalize("hello", "world");
+    }
+
+    public void finalize(String ... args) {}
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>variable name false-positive test</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    private int finalize;
+
+    public void bar() {
+        finalize++;
+        return finalize;
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>super.finalize in constructor false-negative test</description>
+        <expected-problems>1</expected-problems>
```

Comment:
Please add here additional this:

```
<expected-linenumbers>4</expected-linenumbers>
```

That way, we assert also, on which line we report the violation. Thanks!

Reply:
Thank you! Added in 01e48c8.